From ab80cd6e9a414b23f09bdd12e521f573223d9879 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 20 Aug 2018 17:32:44 +0800
Subject: [PATCH 4527/5242] MLK-19413-16 gpu: imx: dpu: tcon: Add
 tcon_is_master/slave() helpers support

commit  9165614a51e7b3413f07b606704c34838d305142 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds tcon_is_master/slave() helpers support so that
callers may know if a tcon is a master or slave tcon.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-tcon.c |   12 ++++++++++++
 include/video/dpu.h            |    2 ++
 2 files changed, 14 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-tcon.c b/drivers/gpu/imx/dpu/dpu-tcon.c
index 4c87cd1..cdc14b0 100644
--- a/drivers/gpu/imx/dpu/dpu-tcon.c
+++ b/drivers/gpu/imx/dpu/dpu-tcon.c
@@ -190,6 +190,18 @@ void tcon_cfg_videomode(struct dpu_tcon *tcon, struct drm_display_mode *m)
 }
 EXPORT_SYMBOL_GPL(tcon_cfg_videomode);
 
+bool tcon_is_master(struct dpu_tcon *tcon)
+{
+	return tcon->id == 0;
+}
+EXPORT_SYMBOL_GPL(tcon_is_master);
+
+bool tcon_is_slave(struct dpu_tcon *tcon)
+{
+	return tcon->id == 1;
+}
+EXPORT_SYMBOL_GPL(tcon_is_slave);
+
 struct dpu_tcon *dpu_tcon_get(struct dpu_soc *dpu, int id)
 {
 	struct dpu_tcon *tcon;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 40d796c..e1177e4 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -670,6 +670,8 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 int tcon_set_fmt(struct dpu_tcon *tcon, u32 bus_format);
 void tcon_set_operation_mode(struct dpu_tcon *tcon);
 void tcon_cfg_videomode(struct dpu_tcon *tcon, struct drm_display_mode *m);
+bool tcon_is_master(struct dpu_tcon *tcon);
+bool tcon_is_slave(struct dpu_tcon *tcon);
 struct dpu_tcon *dpu_tcon_get(struct dpu_soc *dpu, int id);
 void dpu_tcon_put(struct dpu_tcon *tcon);
 struct dpu_tcon *dpu_aux_tcon_peek(struct dpu_tcon *tcon);
-- 
1.7.9.5

